app/src/main/java/org/dhis2/data/forms/dataentry/fields/option_set/OptionSetHolder.kt

```diff
@@ -0,0 +1,65 @@
+package org.dhis2.data.forms.dataentry.fields.option_set
+
+import androidx.lifecycle.MutableLiveData
+import io.reactivex.processors.FlowableProcessor
+import org.dhis2.data.forms.dataentry.fields.FormViewHolder
+import org.dhis2.data.forms.dataentry.fields.RowAction
+import org.dhis2.databinding.FormOptionSetSelectorBinding
+import org.dhis2.utils.customviews.OptionSetSelectionView
+import org.jetbrains.annotations.NotNull
+
+class OptionSetHolder(
+    private val formBinding: @NotNull FormOptionSetSelectorBinding,
+    private val processor: FlowableProcessor<RowAction>,
+    private val isBgTransparent: Boolean,
+    currentSelection: MutableLiveData<String>
+) : FormViewHolder(formBinding) {
+
+    private lateinit var viewModel: OptionSetViewModel
+
+    init {
+        this.currentUid = currentSelection
+
+        formBinding.optionSetSelectionView.setOnSelectedOptionListener(
+            object : OptionSetSelectionView.OnSelectedOption {
+                override fun onSelectedOption(optionName: String?, optionCode: String?) {
+                    processor.onNext(RowAction.create(viewModel.uid(), optionCode))
+                }
+
+                override fun onOptionsClear() {
+                    processor.onNext(RowAction.create(viewModel.uid(), null))
+                }
+            }
+        )
+    }
+
+    fun update(viewModel: OptionSetViewModel) {
+        this.viewModel = viewModel
+
+        formBinding.optionSetSelectionView.setOptionsToShow(
+            viewModel.optionsToHide, viewModel.optionsToShow
+        )
+        formBinding.optionSetSelectionView.setObjectStyle(viewModel.objectStyle())
+        formBinding.optionSetSelectionView.setEditable(viewModel.editable() == true)
+        formBinding.optionSetSelectionView.setInitValue(
+            viewModel.value(), viewModel.options(), viewModel.fieldRendering().type()
+        )
+        formBinding.optionSetSelectionView.setLabel(viewModel.label(), viewModel.mandatory())
+        formBinding.optionSetSelectionView.setDescription(viewModel.description())
+
+        /*formBinding.optionSetSelectionView.descriptin.setOnClickListener {
+            CustomDialog(
+                itemView.context,
+                viewModel.label(),
+                viewModel.description() ?: "",
+                itemView.context.getString(R.string.action_close),
+                null,
+                201,
+                null
+            ).show()
+        }*/
+    }
+
+    override fun dispose() {}
```

Comment:
empty?

Reply:
It is a common method. But not in use anymore, so I will remove it.